src/Strategy/Yarn.hs

```diff
@@ -23,26 +24,33 @@ findProjects = walk' $ \dir _ files -> do
     Just lock -> do
       let project =
             YarnProject
-            { yarnDir = dir
-            , yarnLock = lock
-            }
+              { yarnDir = dir
+              , yarnLock = lock
+              }
 
       pure ([project], WalkSkipSome ["node_modules"])
 
 mkProject :: (Has ReadFS sig n, Has Diagnostics sig n) => YarnProject -> DiscoveredProject n
 mkProject project =
   DiscoveredProject
-    { projectType = "yarn",
-      projectBuildTargets = mempty,
-      projectDependencyGraph = const $ getDeps project,
-      projectPath = yarnDir project,
-      projectLicenses = pure []
+    { projectType = "yarn"
+    , projectBuildTargets = mempty
+    , projectDependencyGraph = const $ getDeps project
+    , projectPath = yarnDir project
+    , projectLicenses = pure []
     }
 
 getDeps :: (Has ReadFS sig m, Has Diagnostics sig m) => YarnProject -> m (G.Graphing Dependency)
-getDeps = context "Yarn" . context "Static analysis" . YarnLock.analyze' . yarnLock
+getDeps project = context "Yarn" $ getDepsV1 project <||> getDepsV2 project
```

Comment:
this function can also be written as:

```hs
getDeps :: (Has ReadFS sig m, Has Diagnostics sig m) => YarnProject -> m (G.Graphing Dependency)
getDeps = (<||>) <$> getDepsV1 <*> getDepsV2
```

.. which is really cute but unmaintainable